// connect_dynamodb/after_school.js
const { core } = require('./core')

module.exports = {
  get_item: async (after_school_id) => {
    const response = await core.query('PK = :p_key AND SK = :s_key', {
      ':p_key': '#METADATA#AFTER_SCHOOL',
      ':s_key': 'AFTER_SCHOOL#' +  after_school_id
    })
    return (response.Items && response.Items.length > 0) ? response.Items[0] : {}
  },
  get_all: async () => {
    const response = await core.query('PK = :p_key AND begins_with(SK, :s_key)', {
      ':p_key': '#METADATA#AFTER_SCHOOL',
      ':s_key': 'AFTER_SCHOOL#',
    })
    return response.Items
  },
  put: async (after_school_id, name, number, open_types, children) => {
    return await core.put({
      PK: '#METADATA#AFTER_SCHOOL',
      SK: 'AFTER_SCHOOL#' +  after_school_id,
      Name: name,
      Number: number,
      Config: {
        OpenTypes: open_types
      },
      Children: children,
    })
  },
  post: async (after_school_id, name, number, open_types, children) => {
    return await core.post({
      PK: '#METADATA#AFTER_SCHOOL',
      SK: 'AFTER_SCHOOL#' +  after_school_id,
      Name: name,
      Number: number,
      Config: {
        OpenTypes: open_types
      },
      Children: children,
    })
  },
}
