// connect_dynamodb/core.js
const dev = process.env.DEVELOPMENT == 'local' ? { region: 'localhost', endpoint: 'http://host.docker.internal:8000' } : {};

const AWS = require('aws-sdk')
const dynamo = new AWS.DynamoDB.DocumentClient(dev);

const table_name = process.env.DEVELOPMENT == 'local' ? 'test-table' : process.env.DYNAMODB_TABLE_NAME;

exports.core = {
  query: async (KeyConditionExpression, ExpressionAttributeValues) => {
    return dynamo.query({
      TableName: table_name,
      KeyConditionExpression: KeyConditionExpression,
      ExpressionAttributeValues: ExpressionAttributeValues,
    }).promise();
  },

  put: async (item) => {
    return dynamo.put({
      TableName: table_name,
      Item: item
    }).promise();
  },

  // キー重複を許さない場合はこちら
  post: async (item) => {
    return dynamo.put({
      TableName: table_name,
      Item: item,
      ConditionExpression: 'attribute_not_exists(PK) and attribute_not_exists(SK)',
    }).promise();
  },

  delete: async (Keys) => {
    return dynamo.delete({
      TableName: table_name,
      Key: Keys
    }).promise();
  },
}
